dbt_subprojects/tokens/models/prices_v2/minute/prices_v2_dex_minute_raw.sql

```diff
@@ -0,0 +1,65 @@
+{{ config(
+    schema='prices_v2'
+    , alias = 'dex_minute_raw'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , partition_by = ['date']
+    , incremental_strategy = 'merge'
+    , unique_key = ['blockchain', 'contract_address', 'timestamp']
+    , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.timestamp')]
+    )
+}}
+
+WITH dex_trades_filter_and_unnest as (
+    SELECT
+        d.blockchain,
+        d.token_bought_address as contract_address,
+        d.block_time as timestamp,
+        d.amount_usd/d.token_bought_amount as price,
+        d.amount_usd as volume -- in USD
+    FROM {{ source('dex','trades') }} d
+    INNER JOIN {{ref('prices_trusted_tokens')}} t
+        on t.blockchain = d.blockchain
+        and t.contract_address = d.token_sold_address -- the token traded against is trusted
+    LEFT JOIN {{ref('prices_trusted_tokens')}} anti_t
+        on anti_t.blockchain = d.blockchain
+        and anti_t.contract_address = d.token_bought_address -- the subjected token is already in trusted
+    WHERE d.amount_usd > 0 and token_bought_amount > 0 and token_bought_address is not null
+    and anti_t.contract_address is null
+    {% if is_incremental() %}
+    AND {{ incremental_predicate('d.block_time') }}
+    {% endif %}
+
+    UNION ALL
+
+    SELECT
+        d.blockchain,
+        d.token_sold_address as contract_address,
+        d.block_time as timestamp,
+        d.amount_usd/d.token_sold_amount as price,
+        d.amount_usd as volume -- in USD
+    FROM {{ source('dex','trades') }} d
+    INNER JOIN {{ref('prices_trusted_tokens')}} t
+        on t.blockchain = d.blockchain
+        and t.contract_address = d.token_bought_address -- the token traded against is trusted
+    LEFT JOIN {{ref('prices_trusted_tokens')}} anti_t
+        on anti_t.blockchain = d.blockchain
+        and anti_t.contract_address = d.token_sold_address -- the subjected token is already in trusted
+    WHERE d.amount_usd > 0 and token_sold_amount > 0 and token_sold_address is not null
+    and anti_t.contract_address is null
+    {% if is_incremental() %}
+    AND {{ incremental_predicate('d.block_time') }}
+    {% endif %}
+)
+
+
+SELECT
+    blockchain,
+    contract_address,
+    date_trunc('minute',timestamp) as timestamp,
+    sum(price*volume)/sum(volume) as price, -- vwap
```

Comment:
VWAP price is defined in the following way, we can do a follow up on this if it's complicated to implement.
```
VWAP = (Typical Price * Volume) / Volume
Calculating the typical price from the session's first candle, with the formula (High + Low + Close) / 3
```

Reply:
that calc with typical price is only needed if you have volume stats for a price range (candle), with our data we have volume per individual price point (trade) so we can be more accurate and do it as it's currently implemented. 

Unless I'm missing something?

Reply:
hmm yeah you are probably right. let's keep it like this then.